tests/ttnn/unit_tests/operations/test_linear.py

```diff
@@ -262,3 +262,76 @@ def test_bloom_ff2_linear(device):
     )
 
     assert ttnn.pearson_correlation_coefficient(torch_output, output) >= 0.9992
+
+
+@pytest.mark.parametrize(
+    "batch_sizes,m_size,k_size, n_size, use_bias",
+    (
+        (2, None, 256, 1536, True),
+        (2, None, 1536, 1536, True),
+        (2, None, 2048, 1536, True),
+        (2, 333, 4096, 1536, True),
+        (2, None, 1536, 13824, True),
+        (2, None, 1536, 9216, True),
+        (2, 4096, 1536, 1536, True),
+        (2, 333, 1536, 1536, True),
+        (2, 4096, 1536, 6144, True),
+        (2, 4096, 6144, 1536, True),
+        (2, 333, 1536, 6144, True),
+        (2, 333, 6144, 1536, True),
+        (2, None, 1536, 3072, True),
+        (2, 4096, 1536, 64, True),
+    ),
+)
+def test_sd3_5_medium(
+    batch_sizes,
+    m_size,
+    k_size,
+    n_size,
+    use_bias,
+    *,
+    device,
+):
+    input_shape_a = (batch_sizes, m_size, k_size) if m_size is not None else (batch_sizes, k_size)
```

Comment:
For matmul/linear we always refer to MxKxN. m_size should never be None. batch_sizes can be None.

Where ever you have 2, None, you should have None, 2 and then here if batch_sizes is not None else (m_size, k_size)